src/code/z_fbdemo.c

```diff
@@ -1,20 +1,223 @@
+
+    bzero(this, sizeof(TransitionTile));
+    this->frame = 0;
+    this->cols = cols;
+    this->rows = rows;
+    gridSize = (cols + 1) * (rows + 1);
+    this->vtxData = SystemArena_Malloc(gridSize * sizeof(TransitionTileVtxData));
+    this->vtxFrame1 = SystemArena_Malloc(gridSize * sizeof(Vtx));
+    this->vtxFrame2 = SystemArena_Malloc(gridSize * sizeof(Vtx));
+    this->gfx = SystemArena_Malloc(((cols * 9 + 1) * rows + 2) * sizeof(Gfx));
+
+    if ((this->vtxData == NULL) || (this->vtxFrame1 == NULL) || (this->vtxFrame2 == NULL) || (this->gfx == NULL)) {
+        if (this->vtxData != NULL) {
+            SystemArena_Free(this->vtxData);
+            this->vtxData = NULL;
+        }
+        if (this->vtxFrame1 != NULL) {
+            SystemArena_Free(this->vtxFrame1);
+            this->vtxFrame1 = NULL;
+        }
+        if (this->vtxFrame2 != NULL) {
+            SystemArena_Free(this->vtxFrame2);
+            this->vtxFrame2 = NULL;
+        }
+        if (this->gfx != NULL) {
+            SystemArena_Free(this->gfx);
+            this->gfx = NULL;
+        }
+        return NULL;
+    }
+
+    TransitionTile_InitGraphics(this);
+    TransitionTile_InitVtxData(this);
+    this->frame = 0;
+
+    return this;
+}
+
+void TransitionTile_SetVtx(TransitionTile* this) {
+    s32 row;
+    s32 col;
+    Vtx* vtx;
+
+    for (row = 0; row < this->rows + 1; row++) {
+        for (col = 0; col < this->cols + 1; col++) {
+            vtx = (this->frame == 0) ? this->vtxFrame1 : this->vtxFrame2;
+            (vtx + col + row * (this->cols + 1))->n.ob[0] = (this->vtxData + col + row * (this->cols + 1))->x;
+
+            vtx = (this->frame == 0) ? this->vtxFrame1 : this->vtxFrame2;
+            (vtx + col + row * (this->cols + 1))->n.ob[1] = (this->vtxData + col + row * (this->cols + 1))->y;
+        }
+    }
+}
+
+void TransitionTile_Draw(TransitionTile* this, Gfx** gfxP) {
+    Gfx* gfx = *gfxP;
+
+    gSPDisplayList(gfx++, sTransTileSetupDL);
+    TransitionTile_SetVtx(this);
+    gSPMatrix(gfx++, &this->projection, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_PROJECTION);
+    gSPMatrix(gfx++, &this->modelView, G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPSegment(gfx++, 0xA, this->frame == 0 ? this->vtxFrame1 : this->vtxFrame2);
+    gSPSegment(gfx++, 0xB, this->zBuffer);
+    gSPDisplayList(gfx++, sTransTileSetupDL);
+    gSPDisplayList(gfx++, this->gfx);
+    gDPPipeSync(gfx++);
+    this->frame ^= 1;
+    *gfxP = gfx;
+}
+
+void TransitionTile_Suck(TransitionTile* this) {
+    s32 row;
+    s32 col;
+    f32 diffX;
+    f32 diffY;
+    f32 scale;
+
+    for (row = 0; row < this->rows + 1; row++) {
+        for (col = 0; col < this->cols + 1; col++) {
+            diffX = (this->vtxData + col + row * (this->cols + 1))->x - (this->vtxData + 5 + 4 * (this->cols + 1))->x;
+            diffY = (this->vtxData + col + row * (this->cols + 1))->y - (this->vtxData + 5 + 4 * (this->cols + 1))->y;
+            scale = (SQ(diffX) + SQ(diffY)) / 100.0f;
```

Comment:
```suggestion
            scale = (SQ(diffX) + SQ(diffY)) / SQ(10.0f);
```
Perhaps?

Reply:
It works but I'm not sure. I think I would prefer just seeing the 100, since the `SQ`s of `diffX` and `diffY` aren't being compared to it. Open to other's opinions though.

Reply:
I don't have any hard opinion on this. If I had to choose I guess I would go with the `SQ` one, but the plain 100 is fine to me too

Reply:
Either is fine with me too, not too worries about it

Reply:
Think I'd prefer SQ since everything else is SQ'd